atomic_reactor/plugins/exit_floating_tags.py

```diff
@@ -0,0 +1,98 @@
+"""
+Copyright (c) 2019 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import unicode_literals, absolute_import
+
+import json
+
+from atomic_reactor.constants import PLUGIN_FLOATING_TAGS_KEY, PLUGIN_GROUP_MANIFESTS_KEY
+from atomic_reactor.manifest_util import ManifestUtil
+from atomic_reactor.plugin import ExitPlugin
+from atomic_reactor.util import get_floating_images, get_unique_images
+
+
+class FloatingTagsPushPlugin(ExitPlugin):
+    """
+    Push floating tags to registry
+    """
+
+    key = PLUGIN_FLOATING_TAGS_KEY
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, registries=None):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param registries: dict, keys are docker registries, values are dicts containing
+                           per-registry parameters.
+                           Params:
+                            * "secret" optional string - path to the secret, which stores
+                              login and password for remote registry
+        """
+        super(FloatingTagsPushPlugin, self).__init__(tasker, workflow)
+        self.manifest_util = ManifestUtil(workflow, registries, self.log)
+
+    def make_manifest(self, registry, repo, digest):
+        return "{registry}/{repo}@{digest}".format(**vars())
+
+    def add_floating_tags(self, session, manifest_list_data, floating_images):
+        manifest_list = manifest_list_data.get("list_json", {})
+        manifest_digests = manifest_list_data.get("digests", {})
+        list_type = manifest_list["mediaType"]
+        manifest_json = json.dumps(manifest_list, indent=4)
```

Comment:
We don't want to be re-creating the bytes here, we need them to be exactly as they were when we originally stored it.